ext/random/random.c

```diff
@@ -470,7 +470,7 @@ PHPAPI int php_random_bytes(void *bytes, size_t size, bool should_throw)
 	/* Defer to CryptGenRandom on Windows */
 	if (php_win32_get_random_bytes(bytes, size) == FAILURE) {
 		if (should_throw) {
-			zend_throw_exception(zend_ce_exception, "Could not gather sufficient random data", 0);
+			zend_throw_exception(zend_ce_exception, "BCryptGenRandom failed", 0);
```

Comment:
I think this is too specific. While I don't think this exception will ever be raised, if it was, the user would go "WTF? What is BCryptGenRandom()? I didn't call that anywhere." It might make more sense to additionally report the actual error code, but in that case `php_win32_get_random_bytes()` would need to return that. That would be a (minor?) BC break, so I'm not sure if it's actually worth doing this.

Reply:
How about:

```suggestion
			zend_throw_exception(zend_ce_exception, "Failed to retrieve randomness from the operating system (BCryptGenRandom)", 0);
```

and likewise for the Apple 'CCRandomGenerateBytes'?

Reply:
I would say yes